src/Apps/Artist/Routes/Overview/state.ts

```diff
@@ -71,15 +114,30 @@ export class FilterState extends Container<State> {
     }
   }
 
-  setPage(page, mediator) {
+  previousQueryString = ""
+  pushHistory() {
+    const currentQueryString = urlFragmentFromState(this.state)
+    // PriceRangeFilter's onAfterChange event fires twice; this ensures
+    //   we only push that history event once.
+    if (this.previousQueryString !== currentQueryString) {
+      window.history.pushState(
+        {},
+        null,
+        `${window.location.pathname}?${currentQueryString}`
+      )
+      this.previousQueryString = currentQueryString
+    }
+  }
```

Comment:
Should `pushHistory` exist on the `State` container? Since it's not explicitly manipulating state, seems like it could exist just outside of the class (similar to `urlFragmentFromState`), and below be called without need for `this`.

Reply:
This def does seem like the right approach though -- keeps the little engine spinning outside of view 😄

Reply:
The advantages we get out of putting it there are (1) we don't need to pass in `this.state` every time we call it (admittedly a #minor advantage), and (2) to [avoid double-stuffing the browser history](https://github.com/artsy/reaction/pull/2392/files/f3dabb884d616c5c42bbd47b1432dc2a9456b8ca#r282675767), `pushHistory` needs to maintain state of what it just pushed to the history.

Reply:
Er...point 2 is incomplete. It's an advantage in the sense that we'd need to make `pushHistory` maintain state _anyway_, so why not use the mechanism we already have to maintain state?

Reply:
Sounds good!